src/inmanta/agent/handler.py

```diff
@@ -655,6 +655,51 @@ def do_changes(self, ctx: HandlerContext, resource: resources.Resource, changes:
         """
         raise NotImplementedError()
 
+    def deploy(
+        self,
+        ctx: HandlerContext,
+        resource: resources.Resource,
+        requires: Dict[ResourceIdStr, ResourceState],
+        dry_run: bool = False,
+    ) -> None:
+        """
+        This method is always be called by the agent, even when one of the requires of the given resource
+        failed to deploy. The default implementation of this method will deploy the given resource when all its
+        requires were deployed successfully. Override this method if a different condition determines whether the
+        resource should deploy.
+
+        :param ctx: Context object to report changes and logs to the agent and server.
+        :param resource: The resource to deploy
+        :param requires: The dependencies of the resource `resource`
+        :param dry_run: True will only determine the required changes but will not execute them.
+        """
+        def _call_resource_did_dependency_change() -> Result:
```

Comment:
Isn't the return type here a future of result? (there is no async or await, but resource_did_dependency_change is async i think?)

Reply:
Updated.